### PR TITLE
[MM-58515] Migrate tooltips of "components/shared_channel_indicator.tsx" to WithTooltip

### PR DESCRIPTION
**Pull request by Aryakoste**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/27231
Jira https://mattermost.atlassian.net/browse/MM-58515
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/shared_channel_indicator.tsx [hunk 1]
```diff
@@ -6,11 +6,10 @@ import {FormattedMessage} from 'react-intl';
 
 import type {ChannelType} from '@mattermost/types/channels';
 
-import OverlayTrigger from 'components/overlay_trigger';
-import Tooltip from 'components/tooltip';
-
 import {Constants} from 'utils/constants';
 
+import WithTooltip from './with_tooltip';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/shared_channel_indicator.tsx:11]: can we use absolute paths here eg `components/with_tooltip` ?

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by M-ZubairAhmed** [comment]
@Aryakoste hows it going with this?

**Comment by M-ZubairAhmed** [comment]
@Aryakoste can you check the failing CI ?

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by Rajat-Dabade — APPROVED** [review]

**Comment by yasserfaraazkhan** [comment]
@M-ZubairAhmed is there a setup required to view shared channel?
can you help with the setup?

**Comment by M-ZubairAhmed** [comment]
> @M-ZubairAhmed is there a setup required to view shared channel? can you help with the setup?

Yes there is here is the docs https://docs.mattermost.com/onboard/shared-channels.html#set-up-shared-channels

**Comment by yasserfaraazkhan** [comment]
@M-ZubairAhmed I've tried this, but see it not working. Can you please set it up for QA? 

Screenshot 2024-06-26 200246


Screenshot 2024-06-26 200127

**Comment by M-ZubairAhmed** [comment]
This is a simple component so skipping QA since it requires setting up Shared channels

**Pull request merged**